### PR TITLE
Implement clap

### PR DESCRIPTION
**Pull request by seaglade**
The program works exactly the same other than now having a help menu and a version command, but adding new flags will now be significantly easier and the main function is cleaner.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -68,14 +82,10 @@ fn decompress<R: Read, W: Write>(r: R, mut w: W) -> Result<()> {
...
-    let first_arg = args.next();
-    let second_arg = args.next();
-    match (first_arg, second_arg) {
-        (None, _) => compress(r, w),
-        (Some(s), None) if s == "-d" => decompress(r, w),
-        _ => unimplemented!()
+    let args = Args::parse();
+    match args.decompress {
+        false => compress(r, w),
+        true => decompress(r, w),
```
Comment (BenWiederhake) [inline src/main.rs]: `if args.decompress` please :)

**Review by BenWiederhake — COMMENTED** [review]
You sneakily added in a lot of formatting-changes – I approve, but maybe in the future let's do that as separate commits.

I know that Rust loves `match`, but please let's use `if` for booleans :P

File: src/main.rs (hunk 1 above)
Comment (seaglade) [reply src/main.rs]: Ah! My bad, VSCode formatted everything automatically 🤣
I'll make it an `if`, but I thought `match` could be good if we add more args. Give me a second...

**Comment by seaglade** [comment]
That force-push was to revert a commit that did a bunch of other stuff, which I'm stashing to PR after this

**Comment by seaglade** [comment]
There we go!

**Pull request merged**